tests/test_source_cli.py

```diff
@@ -2525,3 +2525,38 @@ def settings_customise_sources(
         cfg = CliApp.run(MySettings)
 
         assert cfg.model_dump() == {'foo': 'bar'}
+
+
+def test_cli_aliases_on_flat_object():
+    class Settings(BaseSettings):
+        option: str = Field(default='foo')
+
+        model_config = SettingsConfigDict(cli_aliases={'option2': 'option'})
+
+    assert CliApp.run(Settings, cli_args=['--option2', 'bar']).model_dump() == {'option': 'bar'}
+
+
+def test_cli_aliases_on_nested_object():
+    class Nested(BaseModel):
+        option: str = Field(default='foo')
+
+    class Settings(BaseSettings):
+        nested: Nested = Nested()
+
+        model_config = SettingsConfigDict(cli_aliases={'option2': 'nested.option'})
+
+    assert CliApp.run(Settings, cli_args=['--option2', 'bar']).model_dump() == {'nested': {'option': 'bar'}}
+
+
+def test_cli_aliases_name_collision():
+    class Nested(BaseModel):
+        option: str = Field(default='foo')
+
+    class Settings(BaseSettings):
+        nested: Nested = Nested()
+        option2: str = Field(default='foo2')
+
+        model_config = SettingsConfigDict(cli_aliases={'option2': 'nested.option'})
+
+    with pytest.raises(SettingsError, match='Settings.option2 has multiple aliases: option2 and nested.option'):
+        CliApp.run(Settings, cli_args=['--option2', 'bar'])
```

Comment:
I tested this locally and it works, but maybe worth adding a test for a twice-nested option?

```
class FooSettings(BaseSettings):
    foo_val: str = "foo"

class BarSettings(BaseSettings):
    bar_val: str = "bar"
    foo: FooSettings = FooSettings()


class Settings(BaseSettings):
    model_config = SettingsConfigDict(
        cli_aliases={
            "foo-val": "bar.foo.foo_val",
            "bar-val": "bar.bar_val",
        },
    )
    bar: BarSettings = BarSettings()
```